src/forms/ToggleSwitch.jsx

```diff
@@ -0,0 +1,119 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import cx from 'classnames';
+
+import { C_TEXT_PRIMARY, C_TEXT_SECONDARY, C_TEXT_HINT } from 'swarm-constants/dist/js/colorConstants.js';
+
+import Button from './Button';
+import Icon from '../media/Icon';
+import withToggleControl from '../utils/WithToggleControl';
+
+export const TOGGLE_SWITCH_CLASS = 'toggleSwitch';
+export const TOGGLE_SWITCH_ACTIVE_CLASS = `${TOGGLE_SWITCH_CLASS}--active`;
+export const TOGGLE_SWITCH_DISABLED_CLASS = `${TOGGLE_SWITCH_CLASS}--disabled`;
+
+export const KNOB_CLASS = `${TOGGLE_SWITCH_CLASS}-knob`;
+export const KNOB_ACTIVE_CLASS = `${KNOB_CLASS}--active`;
+export const KNOB_DISABLED_CLASS = `${KNOB_CLASS}--disabled`;
+
+export const LABEL_CLASS = `${TOGGLE_SWITCH_CLASS}-label`;
+
+/**
+ * @module ToggleSwitch
+ */
+class ToggleSwitch extends React.Component {
+	constructor (props) {
+		super(props);
+
+		this.toggle = this.toggle.bind(this);
+	}
+
+	toggle() {
+		!this.props.disabled && this.props.toggleActive();
```

Comment:
The name of this function is confusing to me. I see that comes from `WithToggleControl` so maybe that's not easily changed.

Reply:
It's definitely not the easiest thing to change at this point, but if you thing it's really that confusing, we can open a separate ticket

Reply:
Don't mess with it.